apps/cowswap-frontend/src/modules/usdAmount/apis/getDefillamaUsdPrice.ts

```diff
@@ -0,0 +1,73 @@
+import { FractionUtils } from '@cowprotocol/common-utils'
+import { SupportedChainId } from '@cowprotocol/cow-sdk'
+import { Fraction, Token } from '@uniswap/sdk-core'
+
+import ms from 'ms.macro'
+
+import { fetchWithRateLimit } from 'common/utils/fetch'
+
+import { RateLimitError, UnknownCurrencyError } from './errors'
+
+interface DefillamaUsdQuote {
+  coins: {
+    [chainAndAddress: string]: {
+      price: number
+    }
+  }
+}
+
+export const DEFILLAMA_PLATFORMS: Record<SupportedChainId, string | null> = {
+  [SupportedChainId.MAINNET]: 'ethereum',
+  [SupportedChainId.GNOSIS_CHAIN]: 'xdai',
+  [SupportedChainId.SEPOLIA]: null,
+}
+
+const BASE_URL = 'https://coins.llama.fi/prices/current'
```

Comment:
Maybe we want to also add an env for this, in case we want to introduce a proxy in the future? disregard if you think this is premature

Reply:
Yeah, used just as a fallback now, don't think we need that yet.